pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -349,7 +349,7 @@ stdenv.mkDerivation ({
 
     env = stdenv.mkDerivation {
       name = "interactive-${pname}-${version}-environment";
-      nativeBuildInputs = [ ghcEnv systemBuildInputs ]
+      nativeBuildInputs = [ ghcEnv systemBuildInputs nativeBuildInputs ]
```

Comment:
I left them out on purpose because Cabal needs to know the proper search paths for those libraries to bake those into the Cabal files. Without that information, configured builds won't work outside of a nix-shell environment.

Reply:
nix-shell configures gcc-wrapper magic to find those libraries without an explicit search path. Therefore, the build setup is broken once you leave that shell environment. When you pass --extra-{include,lib}-dir, however, then the configured build works fine inside and outside of the nix-shell environment.

Reply:
> pass --extra-{include,lib}-dir, however

What should I pass this to?

Reply:
You can't pass it anywhere. The best we can do is to create a convenient macro / alias / shell function that users of the environment can use to refer to those flags when invoking `stack` or `cabal` or `tinc` or whatever it is they are invoking.

Reply:
In my opinion these workarounds are required because of the flawed requirement that everything has to work outside of a nix shell after it is configured.
Of course this breaks most workflows from the get-go.

Reply:
The requirement that builds work outside of `nix-shell` is essential for everyone who uses long-running IDEs, like Emacs, to develop multiple projects with different kinds of environments without qutting and re-starting the IDE every time.

Reply:
> like Emacs

I managed to configure my Emacs in a way that it uses the shell environment of the current project for Haskell projects, so I don’t experience that problem.